src/tests/ftest/launch.py

```diff
@@ -701,6 +701,21 @@ def replace_yaml_file(yaml_file, args, tmp_dir):
         with open(yaml_file) as yaml_buffer:
             yaml_data = yaml_buffer.read()
 
+        # Handle missing timeout parameter
+        pattern = r"\n\s*timeout\s*:\s*\d+"
+        if not re.findall(pattern, yaml_data):
+            # Default to 10 minutes, or use the user-specified global timeout
+            # param
+            default_timeout = 600
+            if args.timeout:
+                default_timeout = args.timeout
+
+            insert_timeout_parameter_here = "\n\s*tests\s*:"
```

Comment:
(lint) '\s'. String constant might be missing an r prefix.
(lint) '\s'. String constant might be missing an r prefix.